extension/statistics.js

```diff
@@ -1,36 +1,45 @@
 const STATS_PATTERNS = [{
         name: "stable",
         pattern: null,
+        number: 1,
     },
     {
         name: "nightly",
         pattern: /^\/[^/].*/i,
+        number: 2,
     },
     {
         name: "docs.rs",
         pattern: /^[~@].*/i,
+        number: 3,
     },
     {
         name: "crate",
         pattern: /^!!!.*/i,
+        number: 4,
     },
     {
         name: "attribute",
         pattern: /^#.*/i,
+        number: 5,
     },
     {
         name: "error code",
         pattern: /^`?e\d{2,4}`?$/i,
+        number: 6,
     },
     {
         name: "rustc",
         pattern: /^\/\/.*/i,
+        number: 7,
     },
     {
-        name: 'other',
+        name: "other",
         pattern: /^[>%?]|(v?1\.).*/i,
+        number: 8,
     },
 ];
+const STATS_NUMBER = { 1: "stable", 2: "nightly", 3: "docs.rs", 4: "crate", 5: "attribute", 6: "error code", 7: "rustc", 8: "other" };
```

Comment:
Avoid maintaining the number in two places, how about building the map object from `STATS_PATTER` programmatically?